python/ray/util/sgd/torch/distributed_torch_runner.py

```diff
@@ -283,41 +295,55 @@ def __init__(self, *args, num_cpus=None, num_gpus=None, **kwargs):
 
         # Reserve a local GPU or CPU for the local worker
         # TODO: we should make sure this NEVER dies.
-        self.local_device = "0"
+        self.local_cpu_device = None
+        self.local_cuda_device = "0"
         self._is_set = False
         if num_gpus:
             assert num_gpus == 1, "Does not support multi-gpu workers"
 
-        if not self.is_actor() and num_gpus > 0:
-            self._try_reserve_resources_and_set_cuda()
+        if num_cpus is None:
+            num_cpus = 0
+
+        if num_gpus is None:
+            num_gpus = 0
+
+        if not self.is_actor() and (num_cpus > 0 or num_gpus > 0):
+            self._try_reserve_and_set_resources(num_cpus, num_gpus)
 
         super(LocalDistributedRunner, self).__init__(*args, **kwargs)
 
-    def _try_reserve_resources_and_set_cuda(self, num_cpus, num_gpus):
-        visible_devices = os.environ.get("CUDA_VISIBLE_DEVICES")
-        reserved_gpu_device = reserve_resources(num_cpus, num_gpus)
+    def _try_reserve_and_set_resources(self, num_cpus, num_gpus):
+        visible_cuda_devices = os.environ.get("CUDA_VISIBLE_DEVICES")
+        reserved_cpu_devices, reserved_cuda_device = reserve_resources(
+            num_cpus, num_gpus)
+        self._set_cpu_devices(reserved_cpu_devices)
```

Comment:
I don't think we need to do this, see comments on the test

Reply:
removed